modules/heat_conduction/src/userobjects/FunctorGapFluxModelRadiation.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorGapFluxModelRadiation.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("HeatConductionApp", FunctorGapFluxModelRadiation);
+
+InputParameters
+FunctorGapFluxModelRadiation::validParams()
+{
+  InputParameters params = GapFluxModelBase::validParams();
+  params.addClassDescription("Gap flux model for heat conduction across a gap due to radiation, "
+                             "based on the diffusion approximation.");
+  params.addRequiredParam<MooseFunctorName>("temperature", "The name of the temperature functor");
+  params.addParam<Real>("stefan_boltzmann", 5.670373e-8, "Stefan-Boltzmann constant");
+  params.addRangeCheckedParam<Real>("primary_emissivity",
+                                    1,
+                                    "primary_emissivity>=0 & primary_emissivity<=1",
+                                    "The emissivity of the primary surface");
+  params.addRangeCheckedParam<Real>("secondary_emissivity",
+                                    1,
+                                    "secondary_emissivity>=0 & secondary_emissivity<=1",
+                                    "The emissivity of the secondary surface");
+  return params;
+}
+
+FunctorGapFluxModelRadiation::FunctorGapFluxModelRadiation(const InputParameters & parameters)
+  : GapFluxModelBase(parameters),
+    _T(getFunctor<ADReal>("temperature")),
+    _stefan_boltzmann(getParam<Real>("stefan_boltzmann"))
+{
+  const auto emissivity_primary = getParam<Real>("primary_emissivity");
+  const auto emissivity_secondary = getParam<Real>("secondary_emissivity");
+
+  // Emissivity for plate geometries.
+  _emissivity = emissivity_primary != 0.0 && emissivity_secondary != 0.0
+                    ? 1.0 / emissivity_primary + 1.0 / emissivity_secondary - 1
+                    : 0.0;
```

Comment:
better to call it inverse_emissivity

Reply:
looking at FVInfiniteCylinderRadiativeBC it would not be very hard to also support cylinders? I think some (Cole's) of the cases we care about are cylindrical

Reply:
I second that, cylinders would be good. My application case are PBRs where these are usually cylindrical.

Reply:
damn I was hoping I could get away with just copying the existing non-functor classes. Now you guys are gonna make me think :laughing: Given this feedback, I think I'm going to explore templating these classes such that we can combine the non-functor and functor user object classes into a single template class. Then any physics additions we make will benefit both classes